### PR TITLE
Use WeeAlloc in contracts

### PR DESCRIPTION
**Pull request by krl**

### CODE REVIEW

File: dusk-abi/Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ edition = "2018"
 [dependencies]
 canonical = "0.5"
 canonical_derive = "0.5"
-const-arrayvec = "0.2.1"
+wee_alloc = "0.4.5"
```
Comment (vlopes11) [inline dusk-abi/Cargo.toml:10]: Use `0.4`

File: dusk-abi/src/panic_include.rs [hunk 2]
```diff
@@ -9,6 +9,17 @@
...
 
+// Use `wee_alloc` as the global allocator.
+#[global_allocator]
+static ALLOC: dusk_abi::WeeAlloc = dusk_abi::WeeAlloc::INIT;
+
+#[allow(improper_ctypes_definitions)]
+#[alloc_error_handler]
+#[no_mangle]
+pub extern "C" fn oom(_: ::core::alloc::Layout) -> ! {
+    ::core::intrinsics::abort();
```
Comment (vlopes11) [inline dusk-abi/src/panic_include.rs:20]: Test and document

**Review by vlopes11 — APPROVED** [review]
LGTM, few nits

File: dusk-abi/src/panic_include.rs (hunk 2 above)
Comment (CPerezz) [reply dusk-abi/src/panic_include.rs:20]: This is suggested directly in the `WeeAlloc` documentation. They have some sort of error handling via traps as in `Wasmi` for example. 

See: https://github.com/rustwasm/wee_alloc/blob/master/example/src/lib.rs#L18-L37

We can maybe just copy the comments from there.

**Comment by CPerezz** [comment]
I really like it. Just take into account that last PR merged into WeeAlloc was in Oct 2019. The repo seems active, the mantainers reply to issues and that stuff, but not much development has been happening.  I ignore if this has been already discussed, but just to mention that.

**Pull request merged**